### ISSUE TITLE
False positive for guard clause?

### ISSUE THREAD

**Issue by jacob-carlborg** [bbatsov/rubocop#3555]
I have a case where I believe RuboCop is reporting a false positive:

```ruby
$ cat foo.rb
def foo
  if a
    b = 1
    c = 2
    d = 3
    p b, c, d
  end
end
```

When running RuboCop on the above file, I get the following offense reported:

```
$ rubocop foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:2:3: C: Use a guard clause instead of wrapping the code inside a conditional expression.
  if a
  ^^

1 file inspected, 1 offense detected
```

There are multiple statements inside the above if statement, I don't see how that can be replaced with a guard clause.

I'm running the following version of RuboCop:

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.4, running on ruby 2.1.5 x86_64-darwin15.0)
```

**Comment by jacob-carlborg** [comment]
Here's another example that passes with `0.42.0` but fails with `0.43.0`:

```ruby
def write_assets_manifest
  if Dir.glob(assets_glob).empty?
    FileUtils.mkdir_p(assets_directory)
    FileUtils.touch(assets_manifest_path)
    @assets_manifest_path = assets_manifest_path
  end
end
```

**Comment by savef** [comment]
These look fine to me, it wants you to do this:
```ruby
def foo
  return unless a

  b = 1
  c = 2
  d = 3
  p b, c, d
end
```
And this:
```ruby
def write_assets_manifest
  return unless Dir.glob(assets_glob).empty?

  FileUtils.mkdir_p(assets_directory)
  FileUtils.touch(assets_manifest_path)
  @assets_manifest_path = assets_manifest_path
end
```

**Comment by jacob-carlborg** [comment]
Hmm, ok, I see. Looks like it got a bit smarter between 0.42.0 and 0.43.0.
